flang/lib/Lower/ConvertType.cpp

```diff
@@ -236,6 +236,8 @@ struct TypeBuilder {
       } else if (const Fortran::semantics::DerivedTypeSpec *tySpec =
                      type->AsDerived()) {
         ty = genDerivedType(*tySpec);
+      } else if (type->IsPolymorphic()) {
```

Comment:
It's a little surprising this isn't a subcase of the AsDerived() test immediately above.

Reply:
`AsDerived` returns a pointer to a `DerivedTypeSpec`, whereas `IsPolymorphic` returns a `bool`.  So the result should be the same independent of the order of the two branches.  Even so, I've swapped the order.

I've also added host function name strings to the two new messages to distinguish them from each other.  That should make it easier to determine which one is triggered (first), although both should be valid.  We can readjust those back to indistinguishable messages later if that is preferred for a first release.